app/models/subnet.rb

```diff
@@ -363,6 +364,13 @@ def dns_servers
     [dns_primary, dns_secondary].select(&:present?)
   end
 
+  def masked_subnet_address
+    IPAddr.new(network).mask(mask)
+  rescue StandardError => e
+    logger.info "Unable to mask '#{network}' with '#{mask}': #{e}"
+    IPAddr.new(network)
```

Comment:
Overall I think this is ok. The only thing I'd wonder is if we have some way to validate the subnet so we don't hit this. `normalize_subnet` is called in `before_validation` so it can certainly be hit. I also don't see a validates check on subnet other than presence. Perhaps a future improvement? In that case we probably also want to validate all records in the DB so subnets are valid.

Reply:
Any thoughts on this?